app/src/main/java/cm/aptoide/pt/download/FileDownloadTask.java

```diff
@@ -51,17 +56,35 @@ protected void paused(BaseDownloadTask baseDownloadTask, long soFarBytes, long t
     if (md5Comparator.compareMd5(md5, fileName)) {
       downloadStatus.onNext(new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.COMPLETED,
           FileDownloadManager.PROGRESS_MAX_VALUE, fileType, md5));
-      Log.d("FileDownloader", " Download completed");
+      Log.d(TAG, " Download completed");
     } else {
-      Log.d("FileDownloader", " Download error");
+      Log.d(TAG, " Download error");
       downloadStatus.onNext(
           new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR, 0, fileType, md5));
     }
   }
 
   @Override protected void error(BaseDownloadTask baseDownloadTask, Throwable error) {
-    downloadStatus.onNext(
-        new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR, 0, fileType, md5));
+    error.printStackTrace();
+    if (error instanceof FileDownloadHttpException
+        && ((FileDownloadHttpException) error).getCode() == FILE_NOTFOUND_HTTP_ERROR) {
+      Logger.getInstance()
+          .d(TAG, "File not found error on app: " + md5);
+      downloadStatus.onNext(
+          new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR_FILE_NOT_FOUND, 0,
+              fileType, md5));
+    } else if (error instanceof FileDownloadOutOfSpaceException) {
+      Logger.getInstance()
+          .d(TAG, "Out of space error for the app: " + md5);
+      downloadStatus.onNext(
+          new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR_NOT_ENOUGH_SPACE, 0,
+              fileType, md5));
+    } else {
+      Logger.getInstance()
+          .d(TAG, "Generic error on app: " + md5);
+      downloadStatus.onNext(
+          new FileDownloadTaskStatus(AppDownloadStatus.AppDownloadState.ERROR, 0, fileType, md5));
+    }
```

Comment:
Two things, that may or may not be done:
-Add a new constructor that doesn’t receive a progress
-In each if case create the new FileDownloadTaskStatus and only in the end do the onNext() with that FileDownloadTasStatus

Reply:
Fixed